src/threading/formatters/JSON.h

```diff
@@ -38,18 +45,19 @@ class JSON : public Formatter
 	class NullDoubleWriter : public rapidjson::Writer<rapidjson::StringBuffer>
 		{
 	public:
-		NullDoubleWriter(rapidjson::StringBuffer& stream)
+		[[deprecated("Remove in v7.1 - This is an implementation detail.")]] NullDoubleWriter(
+			rapidjson::StringBuffer& stream)
```

Comment:
I think it'd be better for this to change to be a thin wrapper around json::detail::NullDoubleWriter than to duplicate the code. I don't expect any of this code to change before 7.1 comes out, but it would reduce the maintenance burden in case it does.

Reply:
Not too worried, but I've re-worked it in an additional commit.